app/src/pages/inside/stepPage/modals/editDefectModal/editDefectModal.jsx

```diff
@@ -0,0 +1,340 @@
+        disabled: !externalSystem.length,
+      },
+      {
+        label: intl.formatMessage(messages.saveAndLinkIssueMessage),
+        value: 'Link',
+        onClick: () => {},
+        disabled: !externalSystem.length,
+      },
+      {
+        label: intl.formatMessage(messages.saveAndUnlinkIssueMessage),
+        value: 'Unlink',
+        onClick: () => {},
+        disabled: items.some(
+          (item) => !item.issue.externalSystemIssue || !item.issue.externalSystemIssue.length,
+        ),
+      },
+    ];
+
+    this.state = {
+      ...initialState,
+    };
+  }
+
+  onEditDefects = (closeModal) => {
+    const { items } = this.props.data;
+
+    if (this.checkIfTheDataWasChanged()) {
+      const issues = (this.isBulkEditOperation() &&
+        items.map((item) => {
+          const dataToSend = {
+            test_item_id: item.id,
+            issue: {
+              ...item.issue,
+            },
+          };
+          if (this.state.defectType) {
+            dataToSend.issue.issue_type = this.state.defectType;
+          }
+          if (this.state.markdownValue) {
+            dataToSend.issue.comment = this.state.markdownValue;
+          }
+          return dataToSend;
+        })) || [
+        {
+          test_item_id: items[0].id,
+          issue: {
+            ...items[0].issue,
+            comment: this.state.markdownValue,
+            issue_type: this.state.defectType,
+            ignoreAnalyzer: this.state.ignoreAnalyzer,
+            autoAnalyzed: false,
+          },
+        },
+      ];
+
+      this.fetchEditDefects(issues);
+    }
+
+    closeModal();
+  };
+
+  getReplaceCommentsToAllItemsAvailability = () => {
+    const isReplaceToAllAvailable = localStorage.getItem(REPLACE_COMMENTS_TO_ALL_AVAILABILITY_NAME);
+
+    !isReplaceToAllAvailable &&
+      this.changeReplaceCommentsToAllAvailability(
+        DEFAULT_REPLACE_COMMENTS_TO_ALL_AVAILABILITY_VALUE,
+      );
+    return isReplaceToAllAvailable
+      ? JSON.parse(isReplaceToAllAvailable)
+      : DEFAULT_REPLACE_COMMENTS_TO_ALL_AVAILABILITY_VALUE;
+  };
+
+  checkIfTheDataWasChanged = () => {
+    const { items } = this.props.data;
+    let isDataChanged = false;
+
+    if (this.isBulkEditOperation()) {
+      isDataChanged = Boolean(
+        this.state.defectType ||
+          (this.state.isReplaceCommentsToAllAvailable && this.state.markdownValue),
+      );
+    } else {
+      isDataChanged =
+        this.state.markdownValue !== items[0].issue.comment ||
+        this.state.ignoreAnalyzer !== items[0].issue.ignoreAnalyzer ||
+        this.state.defectType !== items[0].issue.issue_type;
+    }
+    return isDataChanged;
+  };
+
+  fetchEditDefects = (issues) => {
+    const {
+      url,
+      data: { fetchFunc },
+    } = this.props;
+
+    fetch(url, {
+      method: 'put',
+      data: {
+        issues,
+      },
+    }).then(() => {
+      fetchFunc();
+      this.props.showNotification({
+        message: 'SUCCES',
```

Comment:
We have another success message in BB version. Please change it and add translation

Reply:
Please add `catch` section to handle errors

Reply:
Fixed